project/MimaExcludes.scala

```diff
@@ -49,7 +49,34 @@ object MimaExcludes {
 
     //[SPARK-31840] Add instance weight support in LogisticRegressionSummary
     // weightCol in org.apache.spark.ml.classification.LogisticRegressionSummary is present only in current version
-    ProblemFilters.exclude[ReversedMissingMethodProblem]("org.apache.spark.ml.classification.LogisticRegressionSummary.weightCol")
+    ProblemFilters.exclude[ReversedMissingMethodProblem]("org.apache.spark.ml.classification.LogisticRegressionSummary.weightCol"),
```

Comment:
I'm thinking through what if anything breaks here in user code. Are all the subclasses sealed traits? so could any user code have subclassed the old traits in a way that would then not compile now? I don't think so but I'm just going off reading the diff

Reply:
Same general question on the Python side; I'm not as sure about the possibility of breaking something there

Reply:
Before change, the code structure is like this:
```
sealed trait LogisticRegressionSummary extends Serializable
sealed trait LogisticRegressionTrainingSummary extends LogisticRegressionSummary
sealed trait BinaryLogisticRegressionSummary extends LogisticRegressionSummary
sealed trait BinaryLogisticRegressionTrainingSummary extends 
  BinaryLogisticRegressionSummary with LogisticRegressionTrainingSummary
```
So seems OK to me, no users can subclass the old traits.

Python seems OK to me too. I only added a new method. Applications written for old versions of Spark shouldn't be affected by this.